x-pack/legacy/plugins/graph/public/components/field_manager/field_icon.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { ICON_TYPES, palettes, EuiIcon } from '@elastic/eui';
+
+function stringToNum(s: string) {
+  return Array.from(s).reduce((acc, ch) => acc + ch.charCodeAt(0), 1);
+}
+
+function getIconForDataType(dataType: string) {
+  const icons: Partial<Record<string, UnwrapArray<typeof ICON_TYPES>>> = {
```

Comment:
Seems like you're doing this same thing in Lens. Maybe you'll want to figure out a more global service?